Address insensitive language

nrb opened the pull request:
This change will update all links pointing to the `master` branch to instead point to a `main` branch in the Velero codebase.

There are still references to `master` and `blacklist` within the code, based on 3rd party dependencies.

The links to `main` files will be broken until we actually make the switch from `master` to `main` on GitHub; I just want to verify it works now.

ashish-amarnath commented on site/_posts/2019-08-22-announcing-velero-1.1.md:
```diff
@@ -36,7 +36,7 @@ We’ve also made several other enhancements to Velero that should benefit all u
 Users sometimes ask about recommendations for Velero’s resource allocation within their cluster. To help with this concern, we’ve added default resource requirements to the Velero Deployment and restic init containers, along with configurable requests and limits for the restic DaemonSet. All these values can be adjusted if your environment requires it.
 
-We’ve also taken some time to improve Velero for the future by updating the Deployment and DaemonSet to use the apps/v1 API group, which will be the [default in Kubernetes 1.16](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.16.md#action-required-3). This change means that `velero install` and the `velero plugin` commands will require Kubernetes 1.9 or later to work. Existing Ve ...
+We’ve also taken some time to improve Velero for the future by updating the Deployment and DaemonSet to use the apps/v1 API group, which will be the [default in Kubernetes 1.16](https://github.com/kubernetes/kubernetes/blob/main/CHANGELOG-1.16.md#action-required-3). This change means that `velero install` and the `velero plugin` commands will require Kubernetes 1.9 or later to work. Existing Vele ...
```
https://github.com/kubernetes/kubernetes/blob/main/CHANGELOG-1.16.md#action-required-3 is broken

nrb commented on .github/workflows/push.yml:
```diff
@@ -1,4 +1,4 @@
-name: Master CI
+name: Main CI
```
Until this is merged, this is going to be a broken badge on previews, I think.

ashish-amarnath commented on site/_posts/2019-10-10-Velero-v1-1-Stateful-Backup-vSphere.md:
```diff
@@ -29,7 +29,7 @@ Velero version 1.1 provides support to backup applications orchestrated on upstr
 ## Download and Deploy Cassandra
 
-For instructions on how to download and deploy a simple Cassandra StatefulSet, please refer to [this blog post](https://cormachogan.com/2019/06/12/kubernetes-storage-on-vsphere-101-statefulset/). This will show you how to deploy a Cassandra StatefulSet which we can use to do our Stateful aplication backup and restore. The manifests [available here](https://github.com/cormachogan/vsphere-storage-1 ...
+For instructions on how to download and deploy a simple Cassandra StatefulSet, please refer to [this blog post](https://cormachogan.com/2019/06/12/kubernetes-storage-on-vsphere-101-statefulset/). This will show you how to deploy a Cassandra StatefulSet which we can use to do our Stateful aplication backup and restore. The manifests [available here](https://github.com/cormachogan/vsphere-storage-1 ...
```
broken link

ashish-amarnath commented on site/_posts/2020-05-27-CSI-integration.md:
```diff
@@ -27,17 +27,17 @@ As the [Kubernetes CSI Snapshot Beta API](https://kubernetes.io/docs/concepts/st
 This post uses an AKS cluster running Kubernetes `1.17`, with Azure disk CSI driver as an example.
 
-Following instructions to install the Azure disk CSI driver from [here](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/docs/install-csi-driver-master.md) run the below command
+Following instructions to install the Azure disk CSI driver from [here](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main/docs/install-csi-driver-main.md) run the below command
```
broken link

ashish-amarnath commented on site/_posts/2020-05-27-CSI-integration.md:
````diff
@@ -27,17 +27,17 @@ As the [Kubernetes CSI Snapshot Beta API](https://kubernetes.io/docs/concepts/st
 
 ```bash 
-curl -skSL https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/master/deploy/install-driver.sh | bash -s master snapshot -- 
+curl -skSL https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/main/deploy/install-driver.sh | bash -s main snapshot -- 
````
the link and the bash sub-command are not valid

ashish-amarnath commented on site/_posts/2020-05-27-CSI-integration.md:
```diff
@@ -27,17 +27,17 @@ As the [Kubernetes CSI Snapshot Beta API](https://kubernetes.io/docs/concepts/st
-- [`deployment.apps/csi-azuredisk-controller`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/csi-azuredisk-controller.yaml)
-- [`daemonset.apps/csi-azuredisk-node`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/csi-azuredisk-node.yaml)
-- [`deployment.apps/csi-snapshot-controller`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/csi-snapshot-controller.yaml)
+- [`deployment.apps/csi-azuredisk-controller`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main/deploy/csi-azuredisk-controller.yaml)
+- [`daemonset.apps/csi-azuredisk-node`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main/deploy/csi-azuredisk-node.yaml)
+- [`deployment.apps/csi-snapshot-controller`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main/deploy/csi-snapshot-controller.yaml)
```
broken links

ashish-amarnath commented on site/_posts/2020-05-27-CSI-integration.md:
````diff
@@ -85,7 +85,7 @@ parameters:
 ```
 
-NOTE: The above `yaml` was sourced from [StorageClass](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/example/storageclass-azuredisk-csi.yaml) and [VolumeSnapshotClass](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/example/snapshot/storageclass-azuredisk-snapshot.yaml).
+NOTE: The above `yaml` was sourced from [StorageClass](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main/deploy/example/storageclass-azuredisk-csi.yaml) and [VolumeSnapshotClass](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main/deploy/example/snapshot/storageclass-azuredisk-snapshot.yaml).
````
broken links

nrb commented:
@a-mccarthy Thanks!

In terms of the docs site, yeah, I think we should investigate redirecting master to main. In terms of the git branch, I'm not sure how that will work; we can rename our own branches, but I don't think GitHub does redirects for renamed branches, so we'll need to do some research.

Some possible solutions for docs:

 * Get a redirect solution from master to main
 * Leave master docs in place, but remove them from the ToC and put a custom banner in place referencing the name change, and stop updating them.
 * Something else?

Also @tbatard, this will affect #2645, so I want to give you a heads up.

a-mccarthy commented:
Good point about Github, i was referring only to docs, but i'm not sure that Github does redirects for branches either.

I personally like adding a redirect to the docs site, bc it has less overheard than keeping the master docs folder around and adding a warning. Netlify handles redirects in the [config file](https://docs.netlify.com/routing/redirects/#syntax-for-the-netlify-configuration-file)

nrb commented:
Renamed docker master tag

nrb commented:
The redirects for documentation are broken - it looks like the splat directive adds the whole of the path after the domain. Need to figure out how to just redirect the part after `master`/`main`.

a-mccarthy commented on netlify.toml:
```diff
@@ -2,3 +2,9 @@
+
+# Using the * -> :splat rule should redirect anything under master to the matching file under main
+[[redirects]]
+  from = "docs/master/*"
```
@nrb i _think_ you might just need to have a / at the beginning of the to/from values. hoping its that straight forward :)

nrb replied:
Thanks, that looks like it worked!

nrb commented:
@ashish-amarnath @carlisia @a-mccarthy Ok, I think docs are now working and I've updated all of our own scripts, code, and design docs that I'm aware of. I've also rebased it to the latest state of things.

I would appreciate another double check on this, then perhaps we do the switch over on GitHub either tomorrow (July 14) or Thursday (July 15).

Also, @ashish-amarnath or @carlisia, you would both be willing to help with getting the plugins updated, those should be easier, as there's no Netlify docs associated with them. The main replacements I found in here were:

master -> main (only 1 instance of this outside of branch tooling/links)
blacklist -> exclusion or deny list (depending on context)
whitelist -> include or allow list (depending on context)
slave -> secondary or follower (I didn't find any reference to this in Velero's core code, but there were some in the dependencies that we didn't use)

carlisia commented:
I can take the Helm/Azure/GCP. If @ashish-amarnath if overloaded I can take the others as well.

nrb commented:
>I can take the Helm/Azure/GCP.

Thanks! If we have any problems with those, we can discuss them in their own repos.

ashish-amarnath commented on design/csi-snapshots.md:
```diff
@@ -306,16 +306,16 @@ Without these objects, the provider-level snapshots cannot be located in order t
-[11]: https://github.com/heptio/velero/blob/master/pkg/cmd/server/server.go#L471
-[12]: https://github.com/heptio/velero/blob/master/pkg/cmd/util/output/backup_describer.go
-[13]: https://github.com/heptio/velero/blob/master/pkg/cmd/util/output/backup_describer.go#L214
+[7]: https://github.com/heptio/velero/blob/main/pkg/plugin/velero/volume_snapshotter.go#L26
```
Do we want to cleanup the heptio reference here as well?
Not a blocker.

nrb replied:
Might as well, I tried to get ones I saw, but didn't search for them.

carlisia commented:
Does this need to be edited to merge against `main`?

nrb commented:
>Does this need to be edited to merge against main?

@carlisia My intention was to merge into `master` then re-create `main` from that commit. However, I could reverse that order of operations. Recreate `main` so that it's up-to-date, edit the PR to go into `main`, then merge it. That does seem cleaner.

The reason I want to recreate `main` is that it seems cleaner than doing a bunch of merges/rebases since it's a brand new branch.

nrb commented:
@ashish-amarnath - Response inline, monospaced

    https://github.com/nrb/velero/blob/insensitive-language/Makefile#L42 - `Done`
    https://raw.githubusercontent.com/nrb/velero/insensitive-language/CHANGELOG.md - `done`
    Files in https://github.com/nrb/velero/tree/insensitive-language/pkg/backup/testdata/v1 - `I purposefully didn't touch these as they're external CRDs and I wanted to retain their state as-is so we're not changing things.`
    https://github.com/nrb/velero/blob/insensitive-language/site/js/jquery.matchHeight.js#L146 IDK if we should be changing this, but flagging for visibility - `Yeah, here I left it alone cause I have no idea if it'll break anything`.
    Couple of occurrences in https://github.com/nrb/velero/blob/insensitive-language/design/plugin-backup-and-restore-progress-design.md - `Done`
    Some change logs have references to master - `Done, except for links to kubernetes`.

carlisia commented:
> Recreate main so that it's up-to-date, edit the PR to go into main, then merge it. That does seem cleaner.

@nrb I think this is cleaner too. I'll do the same in the other repos.

nrb commented:
Target branch updated.
